pkg/assembler/backends/testing/pkg.go

```diff
@@ -274,6 +473,55 @@ func filterPackageVersion(n *model.PackageName, pkgSpec *model.PkgSpec) *model.P
 	}
 }
 
+func getCollectedPackageQualifiers(qualifieMap map[string]string) []*model.PackageQualifier {
+	qualifiers := []*model.PackageQualifier{}
+	for key, val := range qualifieMap {
+		qualifier := &model.PackageQualifier{
+			Key:   key,
+			Value: val,
+		}
+		qualifiers = append(qualifiers, qualifier)
+
+	}
+	return qualifiers
+}
+
+func getQualifiers(qualifiersSpec []*model.PackageQualifierInputSpec) map[string]string {
+	qualifiersMap := map[string]string{}
+	if qualifiersSpec == nil {
+		return qualifiersMap
+	}
+	for _, kv := range qualifiersSpec {
+		qualifiersMap[kv.Key] = kv.Value
+	}
+	return qualifiersMap
+}
+
+func getQualifiersFromFilter(qualifiersSpec []*model.PackageQualifierSpec) map[string]string {
+	qualifiersMap := map[string]string{}
+	if qualifiersSpec == nil {
+		return qualifiersMap
+	}
+	for _, kv := range qualifiersSpec {
+		qualifiersMap[kv.Key] = *kv.Value
+	}
+	return qualifiersMap
+}
+
+func noMatchQualifiers(filter *model.PkgSpec, v map[string]string) bool {
+	// Allow matching on nodes with no qualifiers
+	if filter.MatchOnlyEmptyQualifiers != nil {
+		if *filter.MatchOnlyEmptyQualifiers && len(v) != 0 {
+			return true
+		}
+	}
+	if filter.Qualifiers != nil {
+		filterQualifiers := getQualifiersFromFilter(filter.Qualifiers)
+		return !reflect.DeepEqual(v, filterQualifiers)
+	}
+	return false
+}
+
 func filterQualifiersAndSubpath(v *model.PackageVersion, pkgSpec *model.PkgSpec) *model.PackageVersion {
```

Comment:
This also is no longer used?

Reply:
keeping for now as others depend on it. Will remove as part of the cleanup

Reply:
added a TODO to cleanup